digraph "unitGraph" {
    "Block 0:
[preds: ] [succs: 1 3 ]
r7 := @parameter0: java.lang.String[];
$r0 = new vm.verifier.VerifyStackForExceptionHandlers;
specialinvoke $r0.<vm.verifier.VerifyStackForExceptionHandlers: void <init>()>();
r1 = $r0;
"
    "Block 1:
[preds: 0 ] [succs: 2 3 ]
specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadGoodClass()>();
"
    "Block 2:
[preds: 1 ] [succs: 4 8 ]
goto [?= specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadBadClass()>()];
"
    "Block 3:
[preds: 0 1 ] [succs: ]
$r5 := @caughtexception;
r8 = $r5;
$r6 = new java.lang.Exception;
specialinvoke $r6.<java.lang.Exception: void <init>(java.lang.String)>(\"FAIL: should be no VerifyError for class A\");
throw $r6;
"
    "Block 4:
[preds: 2 ] [succs: 5 8 ]
specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadBadClass()>();
"
    "Block 5:
[preds: 4 ] [succs: 6 8 ]
$r2 = new java.lang.Exception;
"
    "Block 6:
[preds: 5 ] [succs: 7 8 ]
specialinvoke $r2.<java.lang.Exception: void <init>(java.lang.String)>(\"FAIL: should be a VerifyError for class B\");
"
    "Block 7:
[preds: 6 ] [succs: 8 ]
throw $r2;
"
    "Block 8:
[preds: 2 4 5 6 7 ] [succs: ]
$r3 := @caughtexception;
r9 = $r3;
$r4 = <java.lang.System: java.io.PrintStream out>;
virtualinvoke $r4.<java.io.PrintStream: void println(java.lang.String)>(\"PASS\");
return;
"
    "Block 0:
[preds: ] [succs: 1 3 ]
r7 := @parameter0: java.lang.String[];
$r0 = new vm.verifier.VerifyStackForExceptionHandlers;
specialinvoke $r0.<vm.verifier.VerifyStackForExceptionHandlers: void <init>()>();
r1 = $r0;
"->"Block 1:
[preds: 0 ] [succs: 2 3 ]
specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadGoodClass()>();
";
    "Block 0:
[preds: ] [succs: 1 3 ]
r7 := @parameter0: java.lang.String[];
$r0 = new vm.verifier.VerifyStackForExceptionHandlers;
specialinvoke $r0.<vm.verifier.VerifyStackForExceptionHandlers: void <init>()>();
r1 = $r0;
"->"Block 3:
[preds: 0 1 ] [succs: ]
$r5 := @caughtexception;
r8 = $r5;
$r6 = new java.lang.Exception;
specialinvoke $r6.<java.lang.Exception: void <init>(java.lang.String)>(\"FAIL: should be no VerifyError for class A\");
throw $r6;
";
    "Block 1:
[preds: 0 ] [succs: 2 3 ]
specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadGoodClass()>();
"->"Block 2:
[preds: 1 ] [succs: 4 8 ]
goto [?= specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadBadClass()>()];
";
    "Block 1:
[preds: 0 ] [succs: 2 3 ]
specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadGoodClass()>();
"->"Block 3:
[preds: 0 1 ] [succs: ]
$r5 := @caughtexception;
r8 = $r5;
$r6 = new java.lang.Exception;
specialinvoke $r6.<java.lang.Exception: void <init>(java.lang.String)>(\"FAIL: should be no VerifyError for class A\");
throw $r6;
";
    "Block 2:
[preds: 1 ] [succs: 4 8 ]
goto [?= specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadBadClass()>()];
"->"Block 4:
[preds: 2 ] [succs: 5 8 ]
specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadBadClass()>();
";
    "Block 2:
[preds: 1 ] [succs: 4 8 ]
goto [?= specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadBadClass()>()];
"->"Block 8:
[preds: 2 4 5 6 7 ] [succs: ]
$r3 := @caughtexception;
r9 = $r3;
$r4 = <java.lang.System: java.io.PrintStream out>;
virtualinvoke $r4.<java.io.PrintStream: void println(java.lang.String)>(\"PASS\");
return;
";
    "Block 4:
[preds: 2 ] [succs: 5 8 ]
specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadBadClass()>();
"->"Block 5:
[preds: 4 ] [succs: 6 8 ]
$r2 = new java.lang.Exception;
";
    "Block 4:
[preds: 2 ] [succs: 5 8 ]
specialinvoke r1.<vm.verifier.VerifyStackForExceptionHandlers: void loadBadClass()>();
"->"Block 8:
[preds: 2 4 5 6 7 ] [succs: ]
$r3 := @caughtexception;
r9 = $r3;
$r4 = <java.lang.System: java.io.PrintStream out>;
virtualinvoke $r4.<java.io.PrintStream: void println(java.lang.String)>(\"PASS\");
return;
";
    "Block 5:
[preds: 4 ] [succs: 6 8 ]
$r2 = new java.lang.Exception;
"->"Block 6:
[preds: 5 ] [succs: 7 8 ]
specialinvoke $r2.<java.lang.Exception: void <init>(java.lang.String)>(\"FAIL: should be a VerifyError for class B\");
";
    "Block 5:
[preds: 4 ] [succs: 6 8 ]
$r2 = new java.lang.Exception;
"->"Block 8:
[preds: 2 4 5 6 7 ] [succs: ]
$r3 := @caughtexception;
r9 = $r3;
$r4 = <java.lang.System: java.io.PrintStream out>;
virtualinvoke $r4.<java.io.PrintStream: void println(java.lang.String)>(\"PASS\");
return;
";
    "Block 6:
[preds: 5 ] [succs: 7 8 ]
specialinvoke $r2.<java.lang.Exception: void <init>(java.lang.String)>(\"FAIL: should be a VerifyError for class B\");
"->"Block 7:
[preds: 6 ] [succs: 8 ]
throw $r2;
";
    "Block 6:
[preds: 5 ] [succs: 7 8 ]
specialinvoke $r2.<java.lang.Exception: void <init>(java.lang.String)>(\"FAIL: should be a VerifyError for class B\");
"->"Block 8:
[preds: 2 4 5 6 7 ] [succs: ]
$r3 := @caughtexception;
r9 = $r3;
$r4 = <java.lang.System: java.io.PrintStream out>;
virtualinvoke $r4.<java.io.PrintStream: void println(java.lang.String)>(\"PASS\");
return;
";
    "Block 7:
[preds: 6 ] [succs: 8 ]
throw $r2;
"->"Block 8:
[preds: 2 4 5 6 7 ] [succs: ]
$r3 := @caughtexception;
r9 = $r3;
$r4 = <java.lang.System: java.io.PrintStream out>;
virtualinvoke $r4.<java.io.PrintStream: void println(java.lang.String)>(\"PASS\");
return;
";
}
